client/consensus/slots/src/lib.rs

```diff
@@ -470,6 +471,15 @@ pub enum CheckedHeader<H, S> {
 	Checked(H, S),
 }
 
+
+
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum Error<T> where T: SlotData + Clone + Debug + Send + Sync + 'static {
```

Comment:
Why do we need all this bounds?

Reply:
Putting bounds on the declaration of a struct should be prevented as much as possible. And I don't see any reason here?

Reply:
`Debug + Send + Sync + 'static` are required, otherwise `Error<T>` will not be `Send` or `Sync`. Slot data is there for features. I can remove `Clone`.

Reply:
You only need the bounds on T where you need to have Error Send and Sync

Reply:
Which is the case a few lines down in https://github.com/paritytech/substrate/pull/7472/files/4b752b06c5ddc7ed0bc0d0735651042f3eb24315#diff-2ff27f67b9a711481f3fe7f514fd50c9f63a2bd2852fd47610dbd1a4c758b061R545